geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisHash.java

```diff
@@ -46,19 +52,71 @@
 public class RedisHash extends AbstractRedisData {
   public static final RedisHash NULL_REDIS_HASH = new NullRedisHash();
   private HashMap<ByteArrayWrapper, ByteArrayWrapper> hash;
+  private HashMap<UUID, List<ByteArrayWrapper>> hScanSnapShots;
+  private HashMap<UUID, Long> hScanSnapShotCreationTimes;
+  private ScheduledExecutorService HSCANSnapshotExpirationExecutor = null;
+  private int HSCAN_SNAPSHOTS_EXPIRE_MILLISECONDS;
+  private final int TIME_FOR_SNAPSHOTS_TO_LIVE;
```

Comment:
I think these fields are redundant. This value should also be configurable by the user somehow. Please could you raise a Jira for that.

Reply:
renamed these to 
HSCAN_SNAPSHOTS_EXPIRE_CHECK_FREQUENCY_MILLISECONDS
MINIMUM_MILLISECONDS_FOR_HSCAN_SNAPSHOTS_TO_LIVE 
to clarify intent.  I could still be convinced that they could be the same value, but I feel like there might be cases where users want to have infrequent checks, but want the snapshots to expire soon after creation? If you still wan them combined I won't argue.  I'll put in the ticket to make them configureable